src/App.tsx

```diff
@@ -1,61 +1,107 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const [userId, setUserId] = useState<number>(0);
+  const [title, setTitle] = useState<string>('');
+  const [errorTitle, setErrorTitle] = useState<boolean>(false);
+  const [errorUser, setErrorUser] = useState<boolean>(false);
+
+  const addUser = () => {
+    const id = [...todos].sort(
+      (todo1, todo2) => todo2.id - todo1.id,
+    )[0].id + 1;
+
+    setTodos((prevState) => (
+      [
+        ...prevState,
+        {
+          id,
+          title,
+          completed: false,
+          userId,
+        }]));
+  };
+
+  const handleSubmitButton = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    if (!userId) {
+      setErrorUser(true);
+    }
+
+    if (!title.trim()) {
+      setErrorTitle(true);
+    }
+
+    if (userId && title) {
+      addUser();
+      setUserId(0);
+      setTitle('');
+    }
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.replace(/[^a-z-а-я-0-9-' ']/gi, ''));
```

Comment:
I would recommend to create a constant for regex (ex: const `MY_REGEX = /{{your_regex}}/`)